### PR TITLE
curlpp: add recipe

### PR DESCRIPTION
**Pull request by toge**
### Summary
Changes to recipe:  **curlpp/0.8.1.cci.20240530**

#### Motivation
curlpp is required by [minio-cpp](https://github.com/minio/minio-cpp).
I want to add minio-cpp to cci.

#### Details
https://github.com/jpbarrette/curlpp

---
- [x] Read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md)
- [x] Checked that this PR is not a duplicate: [list of PRs by recipe](https://github.com/conan-io/conan-center-index/discussions/24240)
- [x] Tested locally with at least one configuration using a recent version of Conan

### CODE REVIEW

File: recipes/curlpp/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+    def configure(self):
+        if self.options.shared:
+            self.options.rm_safe("fPIC")
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
+    def requirements(self):
+        self.requires("libcurl/8.9.1")
```
Comment (AbrilRBS) [inline recipes/curlpp/all/conanfile.py:46]: `curlpp/cURLpp.hpp` includes `curl.h`, so this needs to be transitive too

File: recipes/curlpp/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    def validate(self):
+        if self.settings.compiler.get_safe("cppstd"):
+            check_min_cppstd(self, self._min_cppstd)
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.variables["CURLPP_BUILD_SHARED_LIBS"] = self.options.shared
```
Comment (AbrilRBS) [inline recipes/curlpp/all/conanfile.py:57]: The CXX version is hardcoded in https://github.com/jpbarrette/curlpp/blob/master/CMakeLists.txt#L49, could you comment out that line?

**Review by AbrilRBS — COMMENTED** [review]
Thanks! Some comments, looks good otherwise :)

File: recipes/curlpp/all/conanfile.py (hunk 1 above)
Comment (toge) [reply recipes/curlpp/all/conanfile.py]: Fixed.
Thank a lot!

File: recipes/curlpp/all/conanfile.py (hunk 2 above)
Comment (toge) [reply recipes/curlpp/all/conanfile.py:57]: There are three places in CMakeListst.txt
Fixed.

**Review by AbrilRBS — APPROVED** [review]
Thanks a lot! :)

**Review by perseoGI — DISMISSED** [review]
LGTM

**Comment by toge** [comment]
@valgur 
Thank you!
I will fix soon!

**Review by AbrilRBS — APPROVED** [review]
Thanks!

**Review by ErniGH — APPROVED** [review]
LGTM

**Pull request merged**